src/modules/event-stream/use-setup-event-stream.ts

```diff
@@ -0,0 +1,48 @@
+import {useSubscription} from '@atb/api/use-subscription';
+import {useQueryClient} from '@tanstack/react-query';
+import {useCallback} from 'react';
+import {WS_API_BASE_URL} from '@env';
+import {getIdTokenGlobal, useAuthContext} from '../auth';
+import Bugsnag from '@bugsnag/react-native';
+import {handleStreamEvent} from './handle-stream-event';
+import {StreamEvent} from './types';
+import {useFeatureTogglesContext} from '@atb/modules/feature-toggles';
+
+export const useSetupEventStream = () => {
+  const queryClient = useQueryClient();
+  const {abtCustomerId} = useAuthContext();
+  const {isEventStreamEnabled} = useFeatureTogglesContext();
+
+  const url = `${WS_API_BASE_URL}stream/v1`;
+
+  const onMessage = useCallback(
+    (event: WebSocketMessageEvent) => {
+      const message = StreamEvent.safeParse(JSON.parse(event.data));
```

Comment:
What if the JSON fails? We could have a try catch here

Reply:
Good point! Added a util for this kind of thing here: 42fcba11c3a1174654f38f7666f32a0fd11321e2